Corrected webgl scissor clipping

1scaR1 opened the pull request:
Fixes #3204

pandamicro commented on extensions/ccui/layouts/UILayoutWebGLRenderCmd.js:
```diff
@@ -233,7 +261,7 @@
     proto.scissorClippingVisit = function(parentCmd){
         cc.renderer.pushRenderCommand(this._beforeVisitCmdScissor);
-        cc.ProtectedNode.prototype.visit.call(this._node, parentCmd);
+        ccui.ProtectedNode.WebGLRenderCmd.prototype.visit.call(this, parentCmd);
```
I think this should be the line which fixed the issue, so what do other modifications do ?

1scaR1 commented on extensions/ccui/layouts/UILayoutWebGLRenderCmd.js:
```diff
@@ -130,16 +136,38 @@
 
     proto._onBeforeVisitScissor = function(ctx){
-        var clippingRect = this._getClippingRect();
+        this._node._clippingRectDirty = true;
```
This forces node to update clipping rect(this is another issue of not updating clipping rect after moving Layout)

1scaR1 commented:
@pandamicro I'm sorry, the description, i have written is not full.
I have also fixed updating of clipping rect and corrected restoring of clipping state(like in cocos2d-x).
And made logic of `getClippingRect` more clear(like in cocos2d-x)

pandamicro commented:
Thanks for the clarification, can you solve the conflict?

pandamicro commented on extensions/ccui/layouts/UILayout.js:
```diff
@@ -419,38 +419,18 @@ ccui.Layout = ccui.Widget.extend(/** @lends ccui.Layout# */{
-                this._clippingRect.height = finalHeight;
+
+                this._clippingRect.x = Math.max(worldPos.x, parentClippingRect.x);
+                this._clippingRect.y = Math.max(worldPos.y, parentClippingRect.y);
```
Why are you ignoring the anchor point here ? The position is normally its anchor point position.

1scaR1 replied:
I apologize that function `convertToWorldSpace` some above already include anchor point. Or i'm wrong?

pandamicro replied:
Oh, you are right, sorry I missed it

pandamicro commented:
@1scaR1  again, many thanks for the contribution
